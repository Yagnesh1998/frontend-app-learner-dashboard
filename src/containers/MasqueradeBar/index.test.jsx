import React from 'react';
import { shallow } from 'enzyme';
import { formatMessage } from 'testUtils';

import MasqueradeBar from '.';
import hooks from './hooks';

jest.mock('./hooks', () => ({
  useMasqueradeBarData: jest.fn(),
}));

describe('MasqueradeBar', () => {
  const masqueradeMockData = {
    canMasquerade: true,
    isMasquerading: false,
    isMasqueradingFailed: false,
    isMasqueradingPending: false,
    masqueradeInput: '',
    masqueradeErrorMessage: '',
    handleMasqueradeInputChange: jest.fn().mockName('handleMasqueradeInputChange'),
    handleClearMasquerade: jest.fn().mockName('handleClearMasquerade'),
    handleMasqueradeSubmit: jest.fn().mockName('handleMasqueradeSubmit'),
    formatMessage,
  };

  describe('snapshot', () => {
    test('can masquerade', () => {
      hooks.useMasqueradeBarData.mockReturnValueOnce(masqueradeMockData);
      expect(shallow(<MasqueradeBar />)).toMatchSnapshot();
    });
    test('can masquerade with input', () => {
      hooks.useMasqueradeBarData.mockReturnValueOnce({
        ...masqueradeMockData,
        masqueradeInput: 'test',
      });
      expect(shallow(<MasqueradeBar />)).toMatchSnapshot();
    });
    test('cannot masquerade', () => {
      hooks.useMasqueradeBarData.mockReturnValueOnce({
        ...masqueradeMockData,
        canMasquerade: false,
      });
      expect(shallow(<MasqueradeBar />)).toMatchSnapshot();
    });
    test('is masquerading with input', () => {
      hooks.useMasqueradeBarData.mockReturnValueOnce({
        ...masqueradeMockData,
        isMasquerading: true,
        masqueradeInput: 'test',
      });
      expect(shallow(<MasqueradeBar />)).toMatchSnapshot();
    });
    test('is masquerading failed with error', () => {
      hooks.useMasqueradeBarData.mockReturnValueOnce({
        ...masqueradeMockData,
        isMasqueradingFailed: true,
        masqueradeErrorMessage: 'test-error',
      });
      expect(shallow(<MasqueradeBar />)).toMatchSnapshot();
    });
    test('is masquerading pending', () => {
      hooks.useMasqueradeBarData.mockReturnValueOnce({
        ...masqueradeMockData,
        isMasqueradingPending: true,
      });
      expect(shallow(<MasqueradeBar />)).toMatchSnapshot();
    });
  });
});
