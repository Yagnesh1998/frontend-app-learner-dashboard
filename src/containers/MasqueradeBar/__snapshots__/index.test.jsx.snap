// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MasqueradeBar snapshot can masquerade 1`] = `
<div
  className="w-100 shadow-sm px-2"
>
  <Form
    className="masquerade-bar w-100"
  >
    <FormLabel
      className="masquerade-form-label"
      id="masquerade-form-label"
      inline={true}
    >
      <Icon
        src={[MockFunction icons.PersonSearch]}
      />
      View as: 
    </FormLabel>
    <FormGroup
      className="masquerade-form-input"
      isInvalid={false}
    >
      <FormControl
        aria-labelledby="masquerade-form-label"
        floatingLabel="Username or email"
        onChange={[MockFunction handleMasqueradeInputChange]}
        value=""
      />
    </FormGroup>
    <StatefulButton
      className="mr-3"
      disabled={true}
      labels={
        Object {
          "default": "Submit",
        }
      }
      state="default"
      type="submit"
      variant="brand"
    />
  </Form>
</div>
`;

exports[`MasqueradeBar snapshot can masquerade with input 1`] = `
<div
  className="w-100 shadow-sm px-2"
>
  <Form
    className="masquerade-bar w-100"
  >
    <FormLabel
      className="masquerade-form-label"
      id="masquerade-form-label"
      inline={true}
    >
      <Icon
        src={[MockFunction icons.PersonSearch]}
      />
      View as: 
    </FormLabel>
    <FormGroup
      className="masquerade-form-input"
      isInvalid={false}
    >
      <FormControl
        aria-labelledby="masquerade-form-label"
        floatingLabel="Username or email"
        onChange={[MockFunction handleMasqueradeInputChange]}
        value="test"
      />
    </FormGroup>
    <StatefulButton
      className="mr-3"
      disabled={false}
      labels={
        Object {
          "default": "Submit",
        }
      }
      state="default"
      type="submit"
      variant="brand"
    />
  </Form>
</div>
`;

exports[`MasqueradeBar snapshot cannot masquerade 1`] = `""`;

exports[`MasqueradeBar snapshot is masquerading failed with error 1`] = `
<div
  className="w-100 shadow-sm px-2"
>
  <Form
    className="masquerade-bar w-100"
  >
    <FormLabel
      className="masquerade-form-label"
      id="masquerade-form-label"
      inline={true}
    >
      <Icon
        src={[MockFunction icons.PersonSearch]}
      />
      View as: 
    </FormLabel>
    <FormGroup
      className="masquerade-form-input"
      isInvalid={true}
    >
      <FormControl
        aria-labelledby="masquerade-form-label"
        floatingLabel="Username or email"
        onChange={[MockFunction handleMasqueradeInputChange]}
        value=""
      />
      <FormControlFeedback
        hasIcon={false}
        type="invalid"
      />
    </FormGroup>
    <StatefulButton
      className="mr-3"
      disabled={true}
      labels={
        Object {
          "default": "Submit",
        }
      }
      state="default"
      type="submit"
      variant="brand"
    />
  </Form>
</div>
`;

exports[`MasqueradeBar snapshot is masquerading pending 1`] = `
<div
  className="w-100 shadow-sm px-2"
>
  <Form
    className="masquerade-bar w-100"
  >
    <FormLabel
      className="masquerade-form-label"
      id="masquerade-form-label"
      inline={true}
    >
      <Icon
        src={[MockFunction icons.PersonSearch]}
      />
      View as: 
    </FormLabel>
    <FormGroup
      className="masquerade-form-input"
      isInvalid={false}
    >
      <FormControl
        aria-labelledby="masquerade-form-label"
        floatingLabel="Username or email"
        onChange={[MockFunction handleMasqueradeInputChange]}
        value=""
      />
    </FormGroup>
    <StatefulButton
      className="mr-3"
      disabled={true}
      labels={
        Object {
          "default": "Submit",
        }
      }
      state="pending"
      type="submit"
      variant="brand"
    />
  </Form>
</div>
`;

exports[`MasqueradeBar snapshot is masquerading with input 1`] = `
<div
  className="w-100 shadow-sm px-2"
>
  <Form
    className="masquerade-bar w-100"
  >
    <FormLabel
      className="masquerade-form-label"
      inline={true}
    >
      <Icon
        src={[MockFunction icons.PersonSearch]}
      />
      Viewing as: 
    </FormLabel>
    <Chip
      className="masquerade-chip"
      iconAfter={[MockFunction icons.Close]}
      onClick={[MockFunction handleClearMasquerade]}
    >
      test
    </Chip>
  </Form>
</div>
`;
